### PR TITLE
Clean-up for Final DRAFT

### LINKED ISSUE #17: SHALL vs MUST
Section 1.6.4. and throughout.

I know that RFCs typically says "   The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC2119].

My preference (which is what DoD does) is that we not mix words.  Choose and stick with specific words and not switch around.  Despite what the RFCs says, people get confused when different words are used.

Recommend changing 1.6.4 to say that This CP SHALL: 
    Use SHALL and SHALL NOT for  requirements;
    Not use "SHOULD and SHOULD NOT"; and,
    Use MAY when something is optional.

SHOULD and SHOULD NOT are not anything in a policy.  People confuse them with requirements when they are really options (MAY).  

And, yes, I know CA/B forum uses all of the above...  Still not good policy language.

### LINKED ISSUE #33: 6.1.13 Subscriber Key Pair Generation - Debian weak key
Discuss debian weak key checks

> #### 6.1.1.3 Subscriber Key Pair Generation
> The CA SHALL reject a certificate request if the requested Public Key does not meet the requirements set forth in Sections 6.1.5 and 6.1.6 or if it has a known weak Private Key (such as a Debian weak key, see <http://wiki.debian.org/SSLkeys>).

### LINKED ISSUE #46: Certificate Profiles: Section 7.1.4.2.2 Subject Distinguished Name Fields
We have different edits (from docs!) for the Subject Distinguished Name Fields: 

- https://github.com/uspki/policies/blob/master/certificate-policy.md#71422-subject-distinguished-name-fields

Opening issue to _collect_ and _discuss_ the various edits.

### LINKED ISSUE #58: 4.9.1.1  Reasons for revoking a subscriber certificate
Recommend adding "“when the binding between the subject and the subject’s public key defined within a certificate is no longer considered valid (e.g. identifying information in the certificate is no longer valid)”
Item 13 - this is a non-sequitur.  The CA cannot revoke a certificate issued by a subordinate CA.  It can revoke the subordinate CA.  Revocation of CA certificates is handled in Section 4.9.1.2.  
If the intent is that revocation of a subordinate CA for the reason of compromise occurs in 24 hours, then rephrase item 13, or rename the section.

### LINKED ISSUE #98: NSS PKI Wildcard text
NSS PKI has adopted some specific requirements related to approval of wildcard certificates.  As discussed, these need to be considered in this CP.  Here is the specific wildcard related text.

Section 1.3.6.3 (device subscribers) "System or Device SSL or TLS certificates may contain a wildcard designator (*). These certificates are referred to as wildcard certificates in this CP."

Section 3.1.1  "•	For wildcard certificates, the wildcard designator may be included in the Common Name (CN) or SubjectAltName (SAN) or both"

Section 3.1.2 "Wildcard certificates shall not have the wildcard immediately to the left of an agency or organization name.  The name space of the wildcard certificate shall be specific to restrict the use of the certificate to only the system or set of systems designated for a single application or service."

Section 3.1.5 "Where there is a certificate with a wildcard designator in the CN, the PKI shall not issue any other certificate with a CN that overlaps or is a subset of that name space."

Section 3.2.2 "For specific requirements for verification of wildcard certificates, see Section 3.2.3.3."

Section 3.2.3.3 "For wildcard certificates, the RA shall verify that there is a technical justification that necessitates the use of the wildcard."

Section 3.2.3.3 "For wildcard certificates, the RA shall also verify that the entire wildcard namespace is under the control of the PKI Sponsor"

Section 3.2.3.3 "For wildcard certificates, the PKI Sponsor shall be accountable for all copies of the private key."

Section 3.2.3.3 "For wildcard certificates, the RA Officer shall document approval of the technical justification provided by the PKI Sponsor."

Section 4.5.1 "Each NSS PKI member agency that supports wildcard certificates shall develop internal policies and procedures for determining who can be PKI Sponsors for wildcard certificates, managing wildcard namespaces, and for authorizing PKI Sponsors’ control over these namespaces.  NSS PKI member agencies that issue wildcard certificates to commercial services shall ensure that providers document the management and control mechanisms for the wildcard certificates and associated private keys."

Section 6.3.2 table limits cert and key life for wild card certificates to 1 year.

Added a definition "Wildcard Certificate: A System or Device certificate that contains a wildcard designator (*) in the common name (CN) in the Subject field or in the subjectAltName (SAN) extension."

### LINKED ISSUE #142: Removal of Delegated Third Party (all impacted areas)
Initial consensus for US Gov: no Delegated Third Party allowed at all.  Issue opened to track proposed changes and discussions throughout the full CP. 

The following sections require updates: 

- Section 1.3.2 - Registration Authorities
- Section 1.6.1 - Definitions
- Section 3.2.2.4 
- Section 4.2.1
- Section 5.3.7 - _closed in #136_ 
- Section 5.4.1 - _closed in #136_ 
- Section 8.4 - _closed in #140_ 
- Section 8.7 - _closed in #140_ 
- Section 9.8 

In addition:  BRD version 1.4.9 redlined has Ballot 204 (in vote), related to "restricting" Delegated Third Parties: 

- https://cabforum.org/wp-content/uploads/CA-Browser-Forum-BR-1.4.9-redlined.pdf 

We are not allowing entirely.

### LINKED ISSUE #146: Add vulnerability detection and patch management
Add the requirements to Section 6, aligned with US govt and CA/B forum, for: 

- Vulnerability scanning
- Patching (critical and high minimum patching cycles)
- Penetration testing

### LINKED ISSUE #148: 30 days maximum lifetime for secrets
see comments and discussions here: https://github.com/uspki/policies/pull/137

Applies to random values, shared secrets examples, including the domain validation methods: 
- https://github.com/uspki/policies/blob/master/certificate-policy.md#32246-agreed-upon-change-to-website

Add clarifying language to Section 4 lifecycle

### LINKED ISSUE #156: 8.4 Audits and topics
In Section 8.4 as currently drafted, the generic statement includes:

- If a Government CA is required by its Certificate Policy to use a different internal audit scheme...

This **is** the Certificate Policy and there is no current requirement to use a different internal audit scheme. 

Suggest removing this clause; confusing. 

> 
> ## 8.4 Topics covered by assessment
> The CA SHALL undergo an audit in accordance with one of the following schemes:
> 
> 4. If a Government CA is required by its Certificate Policy to use a different internal audit scheme, it MAY use such scheme provided that the audit either (a) encompasses all requirements of one of the above schemes or (b) consists of comparable criteria that are available for public review.

### PR DESCRIPTION
**Pull request by lachellel**
fixes #156 
fixes #146 
fixes #148 
closes #58 (no change)
fixes #17 
closes #33 (no change)
fixes #98 
closes #142 
closes #46

### CODE REVIEW

**Comment by lachellel** [comment]
@konklone

merge?  

separate branch in progress for publication for comment

File: certificate-policy.md [hunk 1]
```diff
@@ -1895,20 +1885,26 @@ c. **Certificate Field:** subject:countryName (OID: 2.5.4.6)
...
+The Subordinate CA Certificate(s) SHALL include the id-kp-serverAuth extended key usage, and the Subordinate CA Certificate(s) SHALL include the Name Constraints X.509v3 extension with constraints on dNSName as follows:
 
 a. For each dNSName in permittedSubtrees, the CA MUST confirm that the Applicant has registered the dNSName or has been authorized by the domain registrant to act on the registrant's behalf in line with the verification practices of section 3.2.2.4.
 The Subordinate CA Certificate MUST include at least one dNSName in permittedSubtrees.  The permittedSubtrees for dNSName MUST be within the constraints of the top-level domains for:
 
  - gov (DotGov)
  - mil (DotMil)
 
-The permittedSubtrees for dNSName MUST NOT contain any other dnsName ranges outside of the the "gov" or "mil" top-level domains.
+The permittedSubtrees for dNSName MUST NOT contain any other dnsName ranges outside of the the DotGov or DotMil top-level domains.
```
Comment (konklone) [inline certificate-policy.md:1907]: Tiny thing (and not going to hold up the merge), but I think `.gov` and `.mil` are the right terms here. `DotGov` and `DotMil` are government program names and terms, but when discussing DNS for a broader audience I think we should refer to it by the technical namespace.

**Comment by konklone** [comment]
Looking good! I left a comment that could be addressed in a subsequent commit. I may just do that and file a PR myself.

File: certificate-policy.md (hunk 1 above)
Comment (konklone) [reply certificate-policy.md:1907]: One concrete example of the confusion: the `DotGov` program also includes `.fed.us`, but the name constraints for this PKI do not include `.fed.us`.

**Comment by konklone** [comment]
> I left a comment that could be addressed in a subsequent commit. I may just do that and file a PR myself.

Actually, there are more references than I expected to `DotGov`. @lachellel, before I go in and look at them, any thoughts?

**Comment by techliaison** [comment]
I agree with Eric that .gov and .mil would be the better terms to use for clarity - but I don't know how many times DotGov was used.

**Comment by konklone** [comment]
I filed #158 to discuss.

**Pull request merged**